candlelight/functional/pchip.py

```diff
@@ -0,0 +1,45 @@
+from typing import Tuple
+
+import torch
+import torch.nn.functional as F
+
+
+def endpoint_slope(delta1: torch.Tensor, delta2: torch.Tensor) -> torch.Tensor:
+    d = (3 * delta1 - delta2) / 2
+    if torch.sign(d) != torch.sign(delta1):
+        d = torch.zeros_like(delta1)
+    elif torch.sign(delta1) != torch.sign(delta2) and torch.abs(d) > torch.abs(
+        3 * delta1
+    ):
+        d = 3 * delta1
+    return d
```

Comment:
What about an `else` branch?

Reply:
What do you mean?
I need to compute `d` before the condition and I always have to return it. What should I do in the `else`?